modules/fluid_properties/src/userobjects/MethaneFluidProperties.C

```diff
@@ -0,0 +1,251 @@
+{
+  return _rho_critical;
+}
+
+Real
+MethaneFluidProperties::rho(Real pressure, Real temperature) const
+{
+  return pressure * _Mch4 / (_R * temperature);
+}
+
+void
+MethaneFluidProperties::rho_dpT(Real pressure, Real temperature, Real & rho, Real & drho_dp, Real & drho_dT) const
+{
+  rho = this->rho(pressure, temperature);
+  drho_dp = _Mch4 / (_R * temperature);
+  drho_dT = - pressure * _Mch4 / (_R * temperature * temperature);
+}
+
+Real
+MethaneFluidProperties::e(Real pressure, Real temperature) const
+{
+  return h(pressure, temperature) - _R * temperature / _Mch4 / 1000.0;
+}
+
+void
+MethaneFluidProperties::e_dpT(Real pressure, Real temperature, Real & e, Real & de_dp, Real & de_dT) const
+{
+  e = this->e(pressure, temperature);
+  Real enthalpy, enthalpy_dp, enthalpy_dT;
+  h_dpT(pressure, temperature, enthalpy, enthalpy_dp, enthalpy_dT);
+  de_dp = enthalpy_dp;
+  de_dT = enthalpy_dT - _R / _Mch4 / 1000.0;
+}
+
+void
+MethaneFluidProperties::rho_e_dpT(Real pressure, Real temperature, Real & rho, Real & drho_dp, Real & drho_dT, Real & e, Real & de_dp, Real & de_dT) const
+{
+  Real density, ddensity_dp, ddensity_dT;
+  rho_dpT(pressure, temperature, density, ddensity_dp, ddensity_dT);
+  rho = density;
+  drho_dp = ddensity_dp;
+  drho_dT = ddensity_dT;
+
+  Real energy, denergy_dp, denergy_dT;
+  e_dpT(pressure, temperature, energy, denergy_dp, denergy_dT);
+  e = energy;
+  de_dp = denergy_dp;
+  de_dT = denergy_dT;
+}
+
+Real
+MethaneFluidProperties::c(Real pressure, Real temperature) const
+{
+  return std::sqrt(gamma(pressure, temperature) * _R * temperature / _Mch4);
+}
+
+Real
+MethaneFluidProperties::cp(Real /*pressure*/, Real temperature) const
+{
+  /// Check the temperatue is in the range of validity (280 K <= T <= 1080 K)
+  if (temperature <= 280.0 || temperature >= 1080.0)
+    mooseError("Temperatue " << temperature << "K out of range (280K, 1080K) in " << name() << ":cp()");
+
+  std::vector<Real> a;
+  if (temperature < 755.0)
+    a = {1.9165258, -1.09269e-3, 8.696605e-6, -5.2291144e-9, 0.0, 0.0, 0.0};
+  else
+    a = {1.04356e1, -4.2025284e-2, 8.849006e-5, -8.4304566e-8, 3.9030203e-11,
+      -7.1345169e-15, 0.0};
+
+  Real specific_heat = 0.0;
+  for (unsigned int i = 0; i < 7; ++i)
+    specific_heat += a[i] * std::pow(temperature, i);
+
+  return specific_heat;
+}
+
+Real
+MethaneFluidProperties::cv(Real pressure, Real temperature) const
+{
+  return cp(pressure, temperature) - _R / _Mch4 / 1000.0;
+}
+
+Real
+MethaneFluidProperties::mu(Real /*density*/, Real temperature) const
+{
+  /// Check the temperatue is in the range of validity (200 K <= T <= 1000 K)
+  if (temperature <= 200.0 || temperature >= 1000.0)
+    mooseError("Temperatue " << temperature << "K out of range (200K, 1000K) in " << name() << ":mu()");
+
+  const Real a[6] = {2.968267e-1, 3.711201e-2, 1.218298e-5, -7.02426e-8, 7.543269e-11,
+    -2.7237166e-14};
+
+  Real viscosity = 0.0;
+  for (unsigned int i = 0; i < 6; ++i)
```

Comment:
I do not like the magic numbers, so `std::vector` would be better. Then you can use `.size()`. Since we accept C++11, you can use this form:
```
std::vector<Real> vec {0.5, 1.33, 2.66};
```